notebooks/utils/CaseClass.py

```diff
@@ -436,6 +478,16 @@ def gen_dataset(
                     f'Time series ends at {ds_timeseries["time_bound"].values[-1,1]}, history files begin at {ds_history["time_bound"].values[0,0]}'
                 )
                 ds = xr.concat([ds_timeseries, ds_history], dim="time", **concat_kwargs)
+                if debug:
+                    print_key_metadata(ds, "xr.concat ds")
+                    print_key_metadata(ds_timeseries, "xr.concat ds_timeseries")
+                    print_key_metadata(ds_history, "xr.concat ds_history")
+                for ds_src in [ds_timeseries, ds_history]:
+                    dict_copy_vals(
+                        ds_src["time"].encoding,
+                        ds["time"].encoding,
+                        ["dtype", "_FillValue", "units", "calendar"],
+                    )
```

Comment:
If I'm reading this code right, `ds` is the concatenation of the two datasets you are looping through as `ds_src` and then you are making sure that the encoding for the `time` variable in `ds` (specifically `dtype`, `_FillValue`, `units`, and `calendar`) match what is in `ds_src` (if those encodings are defined there).

Is it safe to assume that if those encodings are in both `ds_timeseries` and `ds_history` then those values are the same? Or that, if those values differ, that we want to keep the `ds_history` value? I'll continue this thought in the `dict_copy_vals` block of code...